### ISSUE TITLE
azure_rm_aks Error: ManagedClusterIdentity object has no attribute get

### ISSUE THREAD

**Issue by Aaron-von-Awesome** [ansible/ansible_collections_azure#1726]
##### SUMMARY

The `azure_rm_aks` module for azure.azcollection version 2.7.0 is failing on idempotency with the error:

```text
'ManagedClusterIdentity' object has no attribute 'get'
```

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

azure.azcollection:  2.7.0

module: `azure_rm_aks`

##### ANSIBLE VERSION

```paste below
ansible [core 2.16.11]
  config file = /ansible-config/ansible.cfg
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  ansible collection location = /root/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.12.3 (main, Sep 11 2024, 14:17:37) [GCC 13.2.0] (/usr/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```

##### COLLECTION VERSION

ansible-galaxy collection list azure.azcollection
```paste below
# /root/.ansible/collections/ansible_collections
Collection         Version
------------------ -------
azure.azcollection 2.7.0  

# /usr/lib/python3/dist-packages/ansible_collections
Collection         Version
------------------ -------
azure.azcollection 1.19.0 

```

##### CONFIGURATION

```paste below
ANSIBLE_NOCOLOR(/ansible-config/ansible.cfg) = False
ANSIBLE_NOCOWS(/ansible-config/ansible.cfg) = True
CONFIG_FILE() = /ansible-config/ansible.cfg
DEFAULT_LOAD_CALLBACK_PLUGINS(/ansible-config/ansible.cfg) = True
DEFAULT_LOG_PATH(/ansible-config/ansible.cfg) = /ansible-hub/ansible.log
DEFAULT_STDOUT_CALLBACK(/ansible-config/ansible.cfg) = yaml
DEFAULT_VAULT_ID_MATCH(/ansible-config/ansible.cfg) = true
HOST_KEY_CHECKING(/ansible-config/ansible.cfg) = False
INTERPRETER_PYTHON(/ansible-config/ansible.cfg) = /usr/bin/python3
```

##### OS / ENVIRONMENT

Running Ansible to connect to Azure via a custom Docker Container image:

```Dockerfile
ARG IMAGE=24.04

FROM ubuntu:${IMAGE}

ENV ANSIBLE_CONFIG=/ansible-config/ansible.cfg
RUN mkdir /ansible-config/
COPY ./ansible.cfg /ansible-config/ansible.cfg

RUN apt update && \
    apt install -y \
    software-properties-common
RUN add-apt-repository --yes --update ppa:ansible/ansible
RUN apt update && \
    apt install -y \
    ansible \
    curl \
    gcc \
    iproute2 \
    less \
    python3 \
    python3-jmespath \
    python3-kerberos \
    python3-pip \
    python3-requests \
    python3-virtualenv \
    python3-winrm \
    software-properties-common \
    wget && \
    rm -rf /var/lib/apt/lists/*;
RUN rm -rf /usr/lib/python3.11/EXTERNALLY-MANAGED
RUN curl -sL https://aka.ms/InstallAzureCLIDeb | bash
RUN az aks install-cli
RUN ansible-galaxy collection install azure.azcollection --force
RUN pip3 install -r ~/.ansible/collections/ansible_collections/azure/azcollection/requirements.txt --break-system-packages
```

##### STEPS TO REPRODUCE

1. Run Ansible task below to create an AKS Cluster
2. After task completes and AKS Cluster is created, run task again
3. Error produced from `azure_rm_aks`: 
```text
The error was: AttributeError: 'ManagedClusterIdentity' object has no attribute 'get'
```

```yaml
- name: "Provision AKS Cluster --> von-awesome-cluster - k8s version 1.29.7"
  azure.azcollection.azure_rm_aks:
    subscription_id: "<redacted>"
    resource_group: "von-awesome-rg"
    name: "von-awesome-cluster"
    location: "<redacted>"
    kubernetes_version: "1.29.7"
    dns_prefix: "va"
    enable_rbac: true
    api_server_access_profile:
      enable_private_cluster: true
    aad_profile:
      enable_azure_rbac: true
      managed: true
    network_profile:
      service_cidr: "10.111.0.0/16"
      dns_service_ip: "10.111.0.10"
      outbound_type: "loadBalancer"
      network_plugin: "azure"
      network_policy: "azure"
      load_balancer_sku: "standard"
    agent_pool_profiles:
      - name: "sysvanodes"
        count: "1"
        vm_size: "Standard_B4s_v2"
        type: "VirtualMachineScaleSets"
        mode: "System"
        vnet_subnet_id: "<redacted>"
        node_labels:
          type: "system"
      - name: "wkrvanodes"
        count: "2"
        vm_size: "Standard_B4s_v2"
        type: "VirtualMachineScaleSets"
        mode: "User"
        vnet_subnet_id: "<redacted>"
        node_labels:
          type: "worker"
    auto_upgrade_profile:
      node_os_upgrade_channel: "NodeImage" 
      upgrade_channel: "stable" 
    state: present
```

##### EXPECTED RESULTS

I would expect the task to pass due to idempotency.

##### ACTUAL RESULTS

**Full Error:**

```json
{
  "changed": false,
  "module_stderr": "Traceback (most recent call last):\n  File \"/root/.ansible/tmp/ansible-tmp-1727207865.1992385-417680-222872047460951/AnsiballZ_azure_rm_aks.py\", line 107, in <module>\n    _ansiballz_main()\n  File \"/root/.ansible/tmp/ansible-tmp-1727207865.1992385-417680-222872047460951/AnsiballZ_azure_rm_aks.py\", line 99, in _ansiballz_main\n    invoke_module(zipped_mod, temp_path, ANSIBALLZ_PARAMS)\n  File \"/root/.ansible/tmp/ansible-tmp-1727207865.1992385-417680-222872047460951/AnsiballZ_azure_rm_aks.py\", line 47, in invoke_module\n    runpy.run_module(mod_name='ansible_collections.azure.azcollection.plugins.modules.azure_rm_aks', init_globals=dict(_module_fqn='ansible_collections.azure.azcollection.plugins.modules.azure_rm_aks', _modlib_path=modlib_path),\n  File \"<frozen runpy>\", line 226, in run_module\n  File \"<frozen runpy>\", line 98, in _run_module_code\n  File \"<frozen runpy>\", line 88, in _run_code\n  File \"/tmp/ansible_azure.azcollection.azure_rm_aks_payload_dlojqtke/ansible_azure.azcollection.azure_rm_aks_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_aks.py\", line 1572, in <module>\n  File \"/tmp/ansible_azure.azcollection.azure_rm_aks_payload_dlojqtke/ansible_azure.azcollection.azure_rm_aks_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_aks.py\", line 1568, in main\n  File \"/tmp/ansible_azure.azcollection.azure_rm_aks_payload_dlojqtke/ansible_azure.azcollection.azure_rm_aks_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_aks.py\", line 1068, in __init__\n  File \"/tmp/ansible_azure.azcollection.azure_rm_aks_payload_dlojqtke/ansible_azure.azcollection.azure_rm_aks_payload.zip/ansible_collections/azure/azcollection/plugins/module_utils/azure_rm_common.py\", line 474, in __init__\n  File \"/tmp/ansible_azure.azcollection.azure_rm_aks_payload_dlojqtke/ansible_azure.azcollection.azure_rm_aks_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_aks.py\", line 1256, in exec_module\n  File \"/tmp/ansible_azure.azcollection.azure_rm_aks_payload_dlojqtke/ansible_azure.azcollection.azure_rm_aks_payload.zip/ansible_collections/azure/azcollection/plugins/modules/azure_rm_aks.py\", line 1543, in update_identity\nAttributeError: 'ManagedClusterIdentity' object has no attribute 'get'\n",
  "module_stdout": "",
  "msg": "MODULE FAILURE\nSee stdout/stderr for the exact error",
  "rc": 1
}
```

**Comment by Fred-sun** [comment]
@Aaron-von-Awesome Can you share the version of 'azure-mgmt-containerservice' in the environment? Thank you!

**Comment by Aaron-von-Awesome** [comment]
> @Aaron-von-Awesome Can you share the version of 'azure-mgmt-containerservice' in the environment? Thank you!

`azure-mgmt-containerservice: 31.0.0`

**Comment by Fred-sun** [comment]
@Aaron-von-Awesome Fixed in #1730
